lib/joi-extension.js

```diff
@@ -0,0 +1,25 @@
+const regexParser = require('regex-parser')
```

Comment:
It looks like this file is specific to the `search` option but it’s named generically, so maybe it should stay inline?

But it’d be better if we could do without this altogether? Perhaps if we keep the `regex` option?

Reply:
Just preparing for if we needed more joi-extensions :sweat:

Reply:
Keeping the regex option, I would still have to coerce the object into regex.
Same goes for the search.

I prefer to have Joi do the object conversion since it provides that option unless you favor otherwise.

Yes, I can validate against strings and convert them after validation.
But why not do it inside the schema validation?

Reply:
Sorry, I wish GitHub reviews would let you comment on a whole file. I think I confused things by not making it clear I didn't mean to refer to this particular `require` statement, more just this whole file.

Reply:
To make sense as a generic standalone file, I think you might need to move the `required()` and `error()` lines back into main schema definition, and leave this file to just handle the type coercion.

Reply:
After toying with schema generation, I decided not to let the schema do type coercion since the generator could not handle that case.